daemon/kmd/wallet/driver/sqlite.go

```diff
@@ -360,50 +362,70 @@ func checkDBError(err error) error {
 	return nil
 }
 
-// CreateWallet ensures that a wallet of the given name/id combo doesn't exist,
-// and initializes a database with the appropriate name.
-func (swd *SQLiteWalletDriver) CreateWallet(name []byte, id []byte, pw []byte, mdk crypto.MasterDerivationKey) error {
+func (swd *SQLiteWalletDriver) claimWalletNameID(name []byte, id []byte) (dbPath string, err error) {
 	// Grab our lock to avoid races with duplicate wallet names/ids
 	swd.mux.Lock()
 	defer swd.mux.Unlock()
 
-	if len(name) > sqliteMaxWalletNameLen {
-		return errNameTooLong
-	}
-
-	if len(id) > sqliteMaxWalletIDLen {
-		return errIDTooLong
+	for _, nameID := range swd.claimedWallets {
+		if bytes.Equal(nameID[0], name) {
+			return "", errSameName
+		}
+		if bytes.Equal(nameID[1], id) {
+			return "", errSameID
+		}
 	}
 
 	// name, id -> "/data/dir/name-id.db"
-	dbPath := swd.nameIDToPath(name, id)
+	dbPath = swd.nameIDToPath(name, id)
 
 	// Ensure the wallet with this filename doesn't already exist, and that we
 	// have permissions to access the wallet directory
-	_, err := os.Stat(dbPath)
+	_, err = os.Stat(dbPath)
 	if !os.IsNotExist(err) {
-		return err
+		return
 	}
 
 	// Ensure a wallet with this name doesn't already exist. swd.mux is
 	// locked above to avoid races here
 	sameNameDBPaths, err := swd.findDBPathsByName(name)
 	if err != nil {
-		return err
+		return
 	}
 	if len(sameNameDBPaths) != 0 {
-		return errSameName
+		return "", errSameName
 	}
 
 	// Ensure a wallet with this id doesn't already exist. As above, we use
 	// swd.mux to avoid races
 	sameIDDBPaths, err := swd.findDBPathsByID(id)
 	if err != nil {
-		return err
+		return
 	}
 	if len(sameIDDBPaths) != 0 {
-		return errSameID
+		return "", errSameID
+	}
+
+	swd.claimedWallets = append(swd.claimedWallets, [][]byte{name, id})
+	return
+}
+
+// CreateWallet ensures that a wallet of the given name/id combo doesn't exist,
+// and initializes a database with the appropriate name.
+func (swd *SQLiteWalletDriver) CreateWallet(name []byte, id []byte, pw []byte, mdk crypto.MasterDerivationKey) error {
+	if len(name) > sqliteMaxWalletNameLen {
+		return errNameTooLong
+	}
+
+	if len(id) > sqliteMaxWalletIDLen {
+		return errIDTooLong
+	}
+
+	dbPath, err := swd.claimWalletNameID(name, id)
+	if err != nil {
+		return err
 	}
+	// TODO? drop the entry in swd.claimedWallets on exit?
```

Comment:
Might be a good idea in case we're being called by a script that makes tons of wallets.